dashboard/backend/handler/api_handler.go

```diff
@@ -158,19 +166,25 @@ func (apiHandler *APIHandler) handleGetTFJobDetail(request *restful.Request, res
 	})
 	if err != nil {
 		log.Warningf("failed to list pods for TFJob %v under namespace %v: %v", name, namespace, err)
-		response.WriteError(http.StatusInternalServerError, err)
+		if newErr := response.WriteError(http.StatusInternalServerError, err); newErr != nil {
+			log.Errorf("Failed to write response: %v", newErr)
+		}
 	} else {
 		log.Infof("successfully listed pods for TFJob %v under namespace %v", name, namespace)
 		tfJobDetail.Pods = pods.Items
-		response.WriteHeaderAndEntity(http.StatusOK, tfJobDetail)
+		if err = response.WriteHeaderAndEntity(http.StatusOK, tfJobDetail); err != nil {
+			log.Errorf("Failed to write response: %v", err)
+		}
 	}
 }
 
 func (apiHandler *APIHandler) handleDeploy(request *restful.Request, response *restful.Response) {
 	clt := apiHandler.cManager.TFJobClient
 	tfJob := new(v1alpha1.TFJob)
 	if err := request.ReadEntity(tfJob); err != nil {
-		response.WriteError(http.StatusBadRequest, err)
+		if newErr := response.WriteError(http.StatusBadRequest, err); newErr != nil {
```

Comment:
nit: how about use err2, If there is a third error, we can use err3 :)

Reply:
SGTM